abd-clam/src/cakes/knn/sieve_v1.rs

```diff
@@ -291,13 +197,13 @@ impl<'a, T: Send + Sync + Copy, U: Number> Grain<'a, T, U> {
     /// Changes pivot point and swaps elements around so that all elements to left
     /// of pivot are less than or equal to pivot and all elements to right of pivot
     /// are greater than pivot.
-    pub fn _partition(grains: &mut [Self], l: usize, r: usize) -> usize {
-        let pivot = (l + r) / 2;
+    fn _partition_once(grains: &mut [Self], l: usize, r: usize) -> usize {
+        let pivot = l + (r - l) / 2;
```

Comment:
Are these two pivots not equivalent? 

l + (r - l)/2 = l + r/2 - l/2 = l/2 + r/2 = (l+r)/2